### PR TITLE
Split unittest for util func to diffrent source. (#61)

### PR DESCRIPTION
**Pull request by zhoonit**
Split util_func unittest to `unittest_util_func.cpp`.

	Build test: [X]Passed [ ]Failed [ ]Skipped
	Run test: [ ]Passed [X]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

9 tests failed. It seems not related to this PR though, those tests fails in the master branch as well.

[  FAILED  ] 9 tests, listed below:
[  FAILED  ] nntrainer_NeuralNetwork.init_01_p
[  FAILED  ] nntrainer_NeuralNetwork.init_08_n
[  FAILED  ] nntrainer_NeuralNetwork.init_011_p
[  FAILED  ] nntrainer_DataBuffer.setDataFile_01_p
[  FAILED  ] nntrainer_DataBuffer.setDataFile_03_p
[  FAILED  ] nntrainer_DataBuffer.setFeatureSize_01_p
[  FAILED  ] nntrainer_DataBuffer.setFeatureSize_02_n
[  FAILED  ] nntrainer_DataBuffer.init_01_p

### CODE REVIEW

File: test/unittest/unittest_util_func.cpp [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+/**
+ * @brief Main gtest
+ */
+int main(int argc, char **argv) {
+  int result = -1;
+
+  testing::InitGoogleTest(&argc, argv);
```
Comment (gichan-jang) [inline test/unittest/unittest_util_func.cpp:205]: Add to handle exceptions thrown by GTest API.
Refer this PR : https://github.com/nnstreamer/nnstreamer/pull/2077

Comment (zhoonit) [reply test/unittest/unittest_util_func.cpp]: Thanks. Fixed.

**Comment by jijoongmoon** [comment]
> [ FAILED ] 9 tests, listed below:
[ FAILED ] nntrainer_NeuralNetwork.init_01_p
[ FAILED ] nntrainer_NeuralNetwork.init_08_n
[ FAILED ] nntrainer_NeuralNetwork.init_011_p
[ FAILED ] nntrainer_DataBuffer.setDataFile_01_p
[ FAILED ] nntrainer_DataBuffer.setDataFile_03_p
[ FAILED ] nntrainer_DataBuffer.setFeatureSize_01_p
[ FAILED ] nntrainer_DataBuffer.setFeatureSize_02_n
[ FAILED ] nntrainer_DataBuffer.init_01_p

is because you didn't put the input of these test properly. If these would be fail due to any reason, you had CI error.
please look into packages/nntrainer.spec.

**Comment by zhoonit** [comment]
> > [ FAILED ] 9 tests, listed below:
> > [ FAILED ] nntrainer_NeuralNetwork.init_01_p
> > [ FAILED ] nntrainer_NeuralNetwork.init_08_n
> > [ FAILED ] nntrainer_NeuralNetwork.init_011_p
> > [ FAILED ] nntrainer_DataBuffer.setDataFile_01_p
> > [ FAILED ] nntrainer_DataBuffer.setDataFile_03_p
> > [ FAILED ] nntrainer_DataBuffer.setFeatureSize_01_p
> > [ FAILED ] nntrainer_DataBuffer.setFeatureSize_02_n
> > [ FAILED ] nntrainer_DataBuffer.init_01_p
> 
> is because you didn't put the input of these test properly. If these would be fail due to any reason, you had CI error.
> please look into packages/nntrainer.spec & see the log.

Thank you for noticing :)

File: test/unittest/meson.build [hunk 2]
```diff
@@ -10,4 +10,13 @@ unittest_nntrainer_internal = executable('unittest_nntrainer_internal',
...
+
+unittest_util_func = executable('unittest_util_func',
+  'unittest_util_func.cpp',
+  dependencies: [unittest_nntrainer_internal_deps],
+  install: get_option('enable-test'),
+  install_dir: application_install_dir
+)
+
 test('unittest_nntrainer_internal', unittest_nntrainer_internal)
+test('unittest_util_func', unittest_util_func)
```
Comment (dongju-chae) [inline test/unittest/meson.build:22]: how about moving `test(...)` to the below of each executable?

Comment (zhoonit) [reply test/unittest/meson.build:21]: Fixed :)

**Pull request merged**